publisher/server/publisher.go

```diff
@@ -113,12 +115,29 @@ func (s *Server) IsPublishableApp(name string) bool {
 		log.Println(err)
 		return false
 	}
-	if version >= latestRunningVersion(s.EtcdClient, appName) {
+
+	if version >= latestRunningVersion(s.EtcdClient, appName) && isPortOpen(hostAndPort) {
```

Comment:
let's separate this call out. IsPublishableApp determines if the container name follows a regex, where IsPortOpen would be useful on its own.